### PR TITLE
signald: 0.13.1 -> 0.14.1 and add service

### PR DESCRIPTION
**Pull request by Flakebi**
###### Motivation for this change
Update signald to 0.14.1: https://gitlab.com/signald/signald/-/releases
Also drop the git-describe-always patch and use `fetchFromGitLab`. It doesn’t change anything visible for me and seems to be better style. `signald -V` (also the version that is currently in master) always prints unversioned. (For reference, the initial pr is #119285.)
I’m running the module with mautrix-signal and it seems to be working fine.

The module was originally copied from https://gitlab.com/coffeetables/myrdd/-/blob/master/modules/signald.nix, but it changed quite a bit from there.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

cc @Pacman99

### CODE REVIEW

**Comment by expipiplus1** [comment]
Nice! Looking forward to using this.

I have a module (locally) for signald, and it was quite tricky to get file
and avatar sharing to work in both directions. Does this handle sending and
receiving files?

I can take a look in detail and try this out, but not for at least a week
I'm afraid. Please ping me if I seem to have forgotten!

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 139472` at 64eb9d96 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>signald</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `fetchgit`.
Near pkgs/applications/networking/instant-messengers/signald/default.nix:1:26:
```
  |
1 | { lib, stdenv, fetchurl, fetchgit, fetchFromGitLab, jre_headless, coreutils, gradle_6, git, perl
  |                          ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Consider adding a comment explaining the purpose of this patch on the line preceeding.
Near pkgs/applications/networking/instant-messengers/signald/default.nix:51:15:
```
   |
51 |   patches = [ ./gradle-plugin.patch ];
   |               ^
```
</li>
  </ul>
</details>

**Comment by Flakebi** [comment]
Good point about sending and receiving files. I added a group and user parameter, which allows running the signald service under the same user as e.g. mautrix-signald. Also removed the fetchgit leftover noticed by the bot.
I tested sending files from signal to matrix and changing room avatars from matrix to signal and that works now.

I’m using signald in combination with the matrix appservices in #114419. The relevant parts of the config are
```nix
  services.matrix-appservices = {
    addRegistrationFiles = true;
    services = {
      signal = {
        format = "mautrix-python";
        package = pkgs.mautrix-signal;
        serviceDependencies = [ "signald.service" ];
        serviceConfig = {
          ReadWritePaths = [ "/var/lib/signald" ];
        };
        settings = {
          homeserver = …;
          appservice = …;
          bridge = …;

          signal = {
            socket_path = "/run/signald/signald.sock";
            outgoing_attachment_dir = "/var/lib/signald/tmp";
          };
        };
      };
    };
  };

  services.signald = {
    enable = true;
    user = "matrix-as-signal";
    group = "matrix-as-signal";
  };
```

It would be nice to have this somewhere in a service, maybe `services.mautrix-signal` once #114419 is merged?

**Review by Pacman99 — APPROVED** [review]
Looks great!, thanks for helping with upstreaming things from myrdd!

File: nixos/modules/services/misc/signald.nix [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+with lib;
+let
+  cfg = config.services.signald;
+  dataDir = "/var/lib/signald";
+  defaultUser = "signald";
+in
+{
+  options.services.signald = {
+    enable = mkEnableOption ''
+      start the signald service with the socket at
```
Comment (Ma27) [inline nixos/modules/services/misc/signald.nix:12]: The description would now be `Whether to enable start the signald service with the socket` which seems odd.

File: nixos/modules/services/misc/signald.nix [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+      /run/signald/signald.sock
+    '';
+
+    user = mkOption {
+      type = types.str;
+      default = defaultUser;
```
Comment (Ma27) [inline nixos/modules/services/misc/signald.nix:18]: I'm currently running it with `DynamicUser`. Why do we need a user like this here?

**Comment by Flakebi** [comment]
Fixed enable description and made socket path configurable.

File: pkgs/applications/networking/instant-messengers/signald/default.nix [hunk 3]
```diff
@@ -45,11 +41,17 @@ let
     forceShare = [ "dummy" ];
     outputHashAlgo = "sha256";
     outputHashMode = "recursive";
-    outputHash = "0w8ixp1l0ch1jc2dqzxdx3ljlh17hpgns2ba7qvj43nr4prl71l7";
+    # Downloaded jars differ by platform
+    outputHash = {
+      x86_64-linux = "/gJFoT+vvdSWr33oI44XiZXlFfyUjtRVB1M6CMzSztM=";
+      aarch64-linux = "v71stMWBbNALasfGAHvsVTBaDOZfpKK3sQrjNJ6FG1A=";
+    }.${stdenv.system} or (throw "Unsupported platform");
```
Comment (Ma27) [inline pkgs/applications/networking/instant-messengers/signald/default.nix:48]: We should restrict `meta.platforms` accordingly.

**Review by Ma27 — APPROVED** [review]

**Comment by Ma27** [comment]
LGTM, thanks! And againl, sorry for delaying this! :)

**Comment by nh2** [comment]
@Flakebi 8 months later, I'm trying to understand what's needed to use the `mautrix-signal` brige on my Matrix homeserver.

Are you using https://gitlab.com/coffeetables/nix-matrix-appservices now? Is that recommended, or are there simpler setups currently?

**Comment by Ma27** [comment]
I can also share the config I currently use (and perhaps document it in the manual or wiki) when I'm back from vacation. Feelnfree to ping me in case I forget about it :)

**Comment by Flakebi** [comment]
Hi, yes, I’m using that module, although not via the repo (which is quite a lot of code!), but I locally include the module from #114419.
This is my config:
```nix
services.matrix-appservices.services = {
      signal = {
        format = "mautrix-python";
        package = pkgs.mautrix-signal;
        port = whatever;
        serviceDependencies = [ "signald.service" ];
        serviceConfig = {
          EnvironmentFile = "/etc/mautrix-signal/secrets";
          ReadWritePaths = [ "/var/lib/signald" ];
        };
        settings = {
          appservice = {
            provisioning.enabled = false;
            database = "postgres://mautrixsignal:$DBPASS@localhost/mautrixsignal?sslmode=disable";
          };

          bridge = {
            inherit permissions;
            relaybot.authless_portals = false;
            encryption = {
              allow = true;
              default = true;
            };
          };

          signal = {
            socket_path = "/run/signald/signald.sock";
            outgoing_attachment_dir = "/var/lib/signald/tmp";
          };
        };
      };
};

services.signald = {
  enable = true;
  user = "matrix-as-signal";
  group = "matrix-as-signal";
};

```

**Pull request merged**